rx/rx-lite.d.ts

```diff
@@ -349,15 +323,15 @@ declare namespace Rx {
 		select<TResult>(selector: (value: T, index: number, source: Observable<T>) => TResult, thisArg?: any): Observable<TResult>;
 		map<TResult>(selector: (value: T, index: number, source: Observable<T>) => TResult, thisArg?: any): Observable<TResult>;	// alias for select
 		pluck<TResult>(prop: string): Observable<TResult>;
-		selectMany<TOther, TResult>(selector: (value: T) => Observable<TOther>, resultSelector?: (item: T, other: TOther) => TResult): Observable<TResult>;
-		selectMany<TOther, TResult>(selector: (value: T) => IPromise<TOther>, resultSelector?: (item: T, other: TOther) => TResult): Observable<TResult>;
+		selectMany<TOther, TResult>(selector: (value: T) => Observable<TOther>, resultSelector: (item: T, other: TOther) => TResult): Observable<TResult>;
+		selectMany<TOther, TResult>(selector: (value: T) => IPromise<TOther>, resultSelector: (item: T, other: TOther) => TResult): Observable<TResult>;
```

Comment:
heh. There are many declarations that could be merged. I focused in merging only those that caused combinatorial explosion.